---
layout: post
title: Sanger Sequencing Training
date: '2023-06-02'
categories: Tutorial
tags: Bioinformatics Molecular Sanger
---

Introduction to Sanger sequencing and training for coral species identification. 

# Software 

We will be using [Geneious Prime software](https://www.geneious.com/), which I obtained through a University of Rhode Island site license. 

# What is Sanger sequencing? 

In order to conduct marker gene analyses, we are extracting DNA from a sample and we are interested if there are nucleotide base differences (single nucleotide polymorphism). We need to identify a region in the genome that we can amplify that is conserved enough to amplify with enough variation to tell the difference between species, genotypes, and populations. 

In Sanger sequencing, we are looking at one sequence at a time. First, in the PCR we will add a primer of the conserved region, DNA polymerase (Taq polymerase able to handle high temperature), the DNA sample, and fluorescent tagged nucleotides mixed with non-tagged nucleotides. We then QC our PCR product on a gel to confirm expected length. 

Next we move to Sanger sequencing, which uses fluorescent tagged nucleotides at the terminating end. The sequences of each length  runs through a capillary, smallest first. The laser beam and excitation then indicates which nucleotide is which and these are then lined up to give you a sequence! 

![](https://www.sigmaaldrich.com/deepweb/assets/sigmaaldrich/marketing/global/images/technical-documents/protocols/genomics/sequencing/sanger-sequencing_steps_process_diagram/sanger-sequencing_steps_process_diagram.png)

![](https://letstalkscience.ca/sites/default/files/styles/width_800px/public/2020-07/capillary_gel_electrophoresis.png?itok=vlzUQLl7)

We then get a chromatogram to tell us what the sequence is! 



