---
layout: post
title: Pocillopora and Porites Species ID Sanger Sequencing and RFLP protocol
date: '2024-03-30'
categories: Protocol
tags: PCR RFLP Molecular Sanger Protocol
---

This post details protocols for Sanger Sequencing and RFLP assays for species identification of *Porites* and *Pocillopora* from gDNA.  

# Overview 

This protocol contains updated and revised protocols as of 20240329 developed in the Putnam Lab and written by A. Huffmyer.   

This protocol uses two successive markers for *Pocillopora* species identification and one marker for *Porites* species identification.  

For *Pocillopora* species identification, this protocol first uses an mtORF marker to distinguish between XXX and XXX. For species identified 

OVERVIEW OF REASONS FOR PROTOCOL AND USING SEQUENCING/RFLPs

# Pocillopora 

This protocol uses Sanger sequencing to identify *Pocillopora* haplotypes. For samples that are identified to Haplotype 1a, an RFLP is required to further distinguish between *P. meandrina* and *P. grandis*.  

## Sample requirements 

This protocol requires extracted and quantified gDNA. We typically preserve samples in [Zymo Research DNA/RNA Shield R1100-250](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/Zymo_r1100-250_dna_rna_shield.pdf) and extract DNA and RNA with the [Zymo Research Quick-DNA™ Miniprep Plus Kit D4069](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/d4068_d4069_quick-dna_miniprep_plus_kit.pdf).  

## Equipment and supplies

Master mix and general reagents:  
 
- [EmeraldAmp GT PCR Master Mix](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/TaKaRa_Emerald_RR320A_DS.pdf)
	- EmeraldAmp GT PCR Master Mix is a loading-dye-added version of EmeraldAmp MAX PCR Master Mix that is optimized for great performance and convenience in both standard and high-throughput PCR applications.
- Loading Dye [NEB 6X Purple Loading Dye NEB Cat # B7024S](XXX)        
- Gel Stain [Biotium GelGreen Nucleic Acid Gel Stain, 10,000X in Water Fisher Cat NC9728313](https://www.fishersci.com/shop/products/gel-green-stain-5ml/NC9728313#?keyword=NC9728313)
- Agarose and 1XTAE buffer 
- 1 Kb DNA ladder [1kb Gel ladder](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/NEB_1kb_Ladder_N3232S.png?raw=true) 
- 100bp DNA ladder [GeneRuler 100 bp DNA Ladder (Thermo FIsher Catalog SM0241)](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/SM0241_GeneRuler_100bp_DNALadder.pdf) 
- Ultrapure nuclease-free water 

mtORF Sanger sequencing primers:  

- Forward primer [mtORF F](XXX)
- Reverse primer [mtORF R](XXX)

POC Histone RFLP primers:  

- Forward primer [POC Histone F](XXX)
- Reverse primer [POC Histone R](XXX)

RFLP restriction enzymes:  

- Restriction Enzyme [XhoI R0146Sk](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/XhoI%20_%20NEB_R0146S.pdf) 
- Restriction Enzyme Buffer [rCutSmart B6004S](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/rCutSmart_Buffer%20_%20NEB_B6004S.pdf) 

Consumables:  

- Pipettes and tips 
- Gel electrophoresis supplies 
- PCR strip tubes
- 1.5 mL tubes 
- Tube racks 
- Freezer boxes 


## mtORF marker - Sanger Sequencing 

xxx

## POC Histone marker - RFLP

OVERVIEW STATEMENT OF MARKER AND RFLP USE 

### Prepare restriction enzyme master mix 

- Prepare the restriction enzyme master mix. First thaw the Xhol RE and rCutSmart buffers on ice. 
- Calculate the volume of restriction enzyme and buffer needed for each reaction. Each reaction requires a 0.1 ratio of rCutSmart 10X buffer to total reaction volume. The reaction requires equal volumes of rCutSmart and Xhol.   

Here are two examples. The first equation is for 15uL sample volume in the reaction (use for dilute samples or weak amplification) or for reduced 7.5uL sample volumes (use for strong amplification to reduce volume used and allow for sufficient volume for troubleshooting).   

``` 
15 µL sample + 1.9 µL rCutSmart buffer + 1.9 µL Xhol = 18.8 µL total volume 
	- 1.9/18.8 = 0.10 

7.5 µL sample + 0.95 µL rCutSmart buffer + 0.95 µL Xhol = 9.4 µL total volume
	- 0.95/7.5 = 0.10  
```

- Multiply the volume of each component by the number of reactions + 10% error. 
- Combine Xhol and rCutSmart to generate a master mix.  

For example:  

TABLE HERE 

### Incubate samples with restriction enzyme 


### Evaluate results using gel electrophoresis 
1. Prepare 

# Porites 

## Equipment and supplies

Master mix and general reagents:  
 
- [EmeraldAmp GT PCR Master Mix](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/TaKaRa_Emerald_RR320A_DS.pdf)
	- EmeraldAmp GT PCR Master Mix is a loading-dye-added version of EmeraldAmp MAX PCR Master Mix that is optimized for great performance and convenience in both standard and high-throughput PCR applications.
- Loading Dye [NEB 6X Purple Loading Dye NEB Cat # B7024S](XXX)        
- Gel Stain [Biotium GelGreen Nucleic Acid Gel Stain, 10,000X in Water Fisher Cat NC9728313](https://www.fishersci.com/shop/products/gel-green-stain-5ml/NC9728313#?keyword=NC9728313)
- Agarose and 1XTAE buffer 
- 1 Kb DNA ladder [1kb Gel ladder](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/NEB_1kb_Ladder_N3232S.png?raw=true) 
- 100bp DNA ladder [GeneRuler 100 bp DNA Ladder (Thermo FIsher Catalog SM0241)](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/images/SM0241_GeneRuler_100bp_DNALadder.pdf) 
- Ultrapure nuclease-free water 

Porites H2 Sanger sequencing primers:  

- Forward primer [H2 F](XXX)
- Reverse primer [H2 R](XXX)

Consumables:  

- Pipettes and tips 
- Gel electrophoresis supplies 
- PCR strip tubes
- 1.5 mL tubes 
- Tube racks 
- Freezer boxes 

## Sanger Sequencing 



xxx

# References 

Protocols were developed based on previous work from [Johnston et al. 2018](https://peerj.com/articles/4355/) and [Tisthammer et al. 2020](https://peerj.com/articles/8550/). 

- Zoe's PCR protocol [is here](https://zdellaert.github.io/ZD_Putnam_Lab_Notebook/PCR-Protocol/) and Species ID protocol [is here](https://github.com/zdellaert/ZD_Putnam_Lab_Notebook/blob/master/protocols/SpeciesID-via-PCR-Sanger-Sequencing.md). 
- Zoe's previous notebook posts using this protocol are [here](https://zdellaert.github.io/ZD_Putnam_Lab_Notebook/E5-PCR-POC-POR-Sites2and3-SpeciesID/) and [here](https://zdellaert.github.io/ZD_Putnam_Lab_Notebook/E5-PCR-POC-POR-Site1-SpeciesID/)
- Hollie's previous notebook posts using this protocol are [here](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/_posts/2023-07-23-PocID.md) and [protocol here](https://github.com/hputnam/Putnam_Lab_Notebook/blob/master/_posts/Species%20ID%20via%20PCR%20and%20Sanger%20Sequencing.md). 
